### ISSUE TITLE
DecorView@[MainActivity] not attached to window manager

### ISSUE THREAD

**Issue by huextrat** [zoontek/react-native-bootsplash#488]
### Bug summary

From 5.0.0 we are seeing this error on some Android devices.
The issue also occurred in 5.0.2.

From our perspective there is randomness, we've had users who've had the problem once but who haven't had the problem since.

```
java.lang.IllegalArgumentException: View=DecorView@b8ebeb6[MainActivity] not attached to window manager
    at android.view.WindowManagerGlobal.findViewLocked(WindowManagerGlobal.java:543)
    at android.view.WindowManagerGlobal.removeView(WindowManagerGlobal.java:450)
    at android.view.WindowManagerImpl.removeViewImmediate(WindowManagerImpl.java:208)
    at android.app.Dialog.dismissDialog(Dialog.java:404)
    at android.app.Dialog.dismiss(Dialog.java:386)
    at com.zoontek.rnbootsplash.RNBootSplashModuleImpl$5$2.onShow(RNBootSplashModuleImpl.java:5)
    at android.app.Dialog$ListenersHandler.handleMessage(Dialog.java:1475)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loopOnce(Looper.java:205)
    at android.os.Looper.loop(Looper.java:294)
    at android.app.ActivityThread.main(ActivityThread.java:8170)
    at java.lang.reflect.Method.invoke(Method.java)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
```

It has also been mentioned here: https://github.com/zoontek/react-native-bootsplash/issues/456#issuecomment-1689342261

Unfornutaley, I don't have any repro it works fine on all my devices 😕, But here are some information about our monitoring:
- Different devices (Samsung, Pixel 6, Pixel 4a, Redmi, Huawei)
- Android 13, 12, 10, 9

Our implementation of RNBootSplash is the basic one, pretty close to the doc.

### Library version

5.0.2

### Environment info

```shell
/
```

### Steps to reproduce

Android only, and it seems to occurred 2–3 seconds after AppStart.

### Reproducible sample code

```js
Cannot reproduce, but should occur with the basic implementation.
```

**Comment by zoontek** [comment]
Are you using `fade: true`?

**Comment by huextrat** [comment]
Yes, we are using the fade `await RNBootSplash.hide({ fade: true });`

**Comment by zoontek** [comment]
I successfully reproduced it by starting app when the screen is locked (and playing with the `Don’t Keep Activities` developer setting).

Just published a new version, `5.0.3` that should solve the issue.
